pkg/config/config_construct_test.go

```diff
@@ -0,0 +1,122 @@
+package config
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_GetConfig(t *testing.T) {
+	tests := []struct {
+		name string
+		cfg  Application
```

Comment:
yeah thats going to be in all of them i just realized, Its not enormous so im going to defer that until later since it doesnt really change the behavior and want to keep progressing on the foundations path